airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/S3StorageOperations.java

```diff
@@ -138,7 +132,13 @@ public String uploadRecordsToBucket(final SerializableBuffer recordsData,
         exceptionsThrown.add(e);
       }
     }
-    throw new RuntimeException(String.format("Exceptions thrown while uploading records into storage: %s", Strings.join(exceptionsThrown, "\n")));
+    final boolean areAllExceptionsAuthExceptions = exceptionsThrown.stream().filter(e -> e instanceof AmazonS3Exception)
```

Comment:
What's the rationale behind checking that all exceptions are authentication issues?

Similar to `BigQueryGcsOperations` can this Auth Error Code be a constant so anybody not sure how come `403` can easily understand the value behind the integer?

Reply:
+1 on making it a constant

Rationale for checking that all errors are auth errors is that theoretically there could be a transient issue with permissions thats resolved on a retry. All issues being auth issues was the behavior I observed when permissions were not set correctly and it seemed safer to make a judgement call on this being a configuration issue when this occurs since matching `any` as opposed to `all` might hide a different issue

Reply:
Got it, that makes sense. I would add that in as a comment if you can since I can see someone looking at this code and coming with a different conclusion and just checking if `any` of the exceptions are Auth and calling it a day

Further more in the `BigQueryException` class they have a section that goes like
```java
  private static final Set<Error> RETRYABLE_ERRORS =
      ImmutableSet.of(
          new Error(500, null), new Error(502, null), new Error(503, null), new Error(504, null));
```

Might be worth considering naming the errors as `AUTH_ERRORS` and placing them into some general utility file like `JdbcUtils.java` so they don't need to be recreated across each destination connector